api/pkg/service/resource/resource.go

```diff
@@ -143,6 +138,27 @@ func (s *service) ByTypeName(ctx context.Context, p *resource.ByTypeNamePayload)
 	return s.resourcesForQuery(q)
 }
 
+// Finds a resource information using it's id
+func (s *service) ByID(ctx context.Context, p *resource.ByIDPayload) (res *resource.Resource, err error) {
+
+	q := s.db.Scopes(withResourceDetails,
+		findByID(p.ID))
+
+	var one model.Resource
+	if err := findOne(q, &one); err != nil {
+		return nil, err
+	}
+
+	res = &resource.Resource{}
```

Comment:
this would be necessary `initResource` create a new object for `resource.Resource`